### PR TITLE
Merge release_candidate with main branch and added a test for the GPU with Ansatz

### PR DESCRIPTION
**Pull request by CassNot**
Here is a proposition to merge `release_candiate` branch with `main` branch:

Differences stem in the reservoir mode (`release_candidate`) and superposition mode (`main`).

**Choices made**
- applied ruff + mypy on `release_candidate` for proper comparison
- removed no_bunching arguments from ansatz
- removed `mlp_model.py` from datasets

**CUDA support**
-  Added QuantumClassifier_withAnsatz class with comprehensive CUDA testing in tests/test_cuda.py

**Test improvements**
- Fixed gradient computation in tests/test_output.py by using MSE loss instead of sum for more stable backpropagation
- Improved dtype handling in tests/test_superposition_state.py with explicit torch.float64 specification
- Added validation for phi_static buffer existence in tests/test_simple_input.py

**Documentation**
- fixed QLLM notebook with better formatting
- fixed variable naming in docs/source/conf.py for consistency

**Test Coverage**
  - All existing tests continue to pass
  - New CUDA tests validate GPU compatibility across different configurations
  - Improved gradient computation stability
  - Enhanced type safety in superposition state handling

### CODE REVIEW

File: tests/test_cuda.py [hunk 1]
```diff
@@ -86,3 +87,192 @@ def test_switch_model_to_cuda():
     assert layer.computation_process.simulation_graph.vectorized_operations[-1][
         0
     ].device == torch.device("cuda", index=0)
+
+
+class QuantumClassifier_withAnsatz(nn.Module):
```
[the remark this answers is not in the document]
Comment (CassNot) [reply tests/test_cuda.py:92]: I am only using this class for testing purposes, not for production
